app/src/main/java/com/example/perfectweatherallyear/util/WorkerHandler.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.perfectweatherallyear.util
+
+import android.content.Context
+import androidx.work.*
+import androidx.work.ListenableWorker.Result.success
+import com.example.perfectweatherallyear.appComponent
+import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.model.HourWeather
+import com.example.perfectweatherallyear.model.Location
+import com.example.perfectweatherallyear.repository.DataResult
+import com.example.perfectweatherallyear.repository.WeatherRepository
+import com.example.perfectweatherallyear.ui.weekWeather.DAYS_NUMBER
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class WorkerHandler(appContext: Context, workerParams: WorkerParameters): Worker(appContext, workerParams) {
+    @Inject
+    lateinit var repository: WeatherRepository
+    lateinit var dayWeatherForNotification: DayWeather
+    lateinit var hourWeatherForNotification:  HourWeather
+
+    init {
+        appContext.appComponent.inject(this)
+    }
+
+    override fun doWork(): Result {
+        GlobalScope.launch {
+            NotificationHandler.createNotification(applicationContext, loadHourWeatherTempForNotification())
```

Comment:
Bad pattern to use as argument complicated functions. Receive result to a var, and then use it as argument.